src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Recognition/Type.hs

```diff
@@ -22,23 +22,25 @@ import Control.Arrow ((&&&))
 import Control.Lens (makeLenses)
 import Data.Aeson (ToJSON)
 import Data.Function (on)
+import Data.HashMap.Strict (HashMap)
+import Data.HashSet (HashSet)
 import Data.Int (Int32)
 import Data.List.NonEmpty qualified as NE
 import Data.Map (Map)
 import Data.Maybe (catMaybes)
 import Data.Ord (Down (Down))
 import Data.Semigroup (Max, Min)
-import Data.Set (Set)
+import Data.Text (Text)
 import GHC.Generics (Generic)
 import Linear (V2 (..))
 import Swarm.Game.Location (Location)
 import Swarm.Game.Scenario.Topography.Area
-import Swarm.Game.Scenario.Topography.Placement (StructureName)
-import Swarm.Game.Scenario.Topography.Structure (NamedGrid)
 import Swarm.Game.Universe (Cosmic, offsetBy)
 import Swarm.Language.Syntax.Direction (AbsoluteDir)
 import Text.AhoCorasick (StateMachine)
 
+type OriginalName = Text
```

Comment:
Could you please add some documentation to `OriginalName` and `NamedOriginal`?

I am not familiar with the code, but I wonder why `StructureName` needed to be switched out.